### PR TITLE
[tensorflow]|[build]|[ec2,ecs,eks] Add E3 dockerfiles for TF2.9

### PR DESCRIPTION
**Pull request by nish21**
*GitHub Issue #, if available:*

Note: If merging this PR should also close the associated Issue, please also add that Issue # to the Linked Issues section on the right. 

### Description

Add E3 dockerfiles for TF2.9

### Tests run
**NOTE: If you are creating a PR for a new framework version, please ensure success of the standard, rc, and efa sagemaker remote tests by updating the dlc_developer_config.toml file:**

- [ ] Revision A: `sagemaker_remote_tests = "standard"`
- [ ] Revision B: `sagemaker_remote_tests = "rc"`
- [ ] Revision C: `sagemaker_remote_tests = "efa"`

**Additionally, please run the sagemaker local tests in at least one revision:**
- [ ] `sagemaker_local_tests = true`

## Label Checklist
- [x] I have added the project label for this PR (*<project_name>* or "Improvement")

## PR Checklist
- [x] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron/graviton] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [x] If the PR changes affects SM test, I've modified dlc_developer_config.toml in my PR branch by setting sagemaker_tests = true and efa_tests = true
- [x] If this PR changes existing code, the change fully backward compatible with pre-existing code. (Non backward-compatible changes need special approval.)
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [x] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON/GRAVITON Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `ei_mode = true`, `neuron_mode = true` or `graviton_mode = true`

#### Benchmark Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `benchmark_mode = true`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by kevinyang8 — DISMISSED** [review]

**Comment by kevinyang8** [comment]
Approved, merge after reverting `dlc_developer_config.toml` to default state

File: test/dlc_tests/container_tests/bin/testTFKerasHVD.py [hunk 1]
```diff
@@ -29,8 +29,8 @@
 if data_type == "AMP":
     # if this code block get excecuted, data type is mixed-precision AKA AMP. If not, it is FP32.
     # When ampere comes and TF32 is available, we need to extend the test to run TF32 as well
-    policy = tf.keras.mixed_precision.experimental.Policy('mixed_float16', 128)
-    tf.keras.mixed_precision.experimental.set_policy(policy)
+    policy = tf.keras.mixed_precision.Policy('mixed_float16')
+    tf.keras.mixed_precision.set_global_policy(policy)
```
Comment (tejaschumbalkar) [inline test/dlc_tests/container_tests/bin/testTFKerasHVD.py:33]: is this backward compatible with previous TF versions?

File: tensorflow/training/docker/2.9/py3/cu112/Dockerfile.gpu [hunk 2]
```diff
@@ -0,0 +1,287 @@
...
+   pip \
+   setuptools
+
+# Some TF tools expect a "python" binary
+RUN ln -s $(which ${PYTHON}) /usr/local/bin/python \
+   && ln -s $(which ${PIP}) /usr/bin/pip
+
+# install PyYAML==5.1.2 to avoid conflict with latest awscli
+# # python-dateutil==2.8.0 to satisfy botocore associated with latest awscli
+RUN ${PIP} install --no-cache-dir -U \
```
Comment (tejaschumbalkar) [inline tensorflow/training/docker/2.9/py3/cu112/Dockerfile.gpu:199]: Are we keeping the pinned versions of pip package as of TF2.8? Any modification required?

File: tensorflow/training/docker/2.9/py3/cu112/Dockerfile.gpu [hunk 3]
```diff
@@ -0,0 +1,287 @@
...
+   && mv /etc/ssh/ssh_config.new /etc/ssh/ssh_config
+
+# Add NGC vars
+ENV TF_AUTOTUNE_THRESHOLD=2
+
+ADD https://raw.githubusercontent.com/aws/deep-learning-containers/master/src/deep_learning_container.py /usr/local/bin/deep_learning_container.py
+
+RUN chmod +x /usr/local/bin/deep_learning_container.py
+
+RUN curl https://aws-dlc-licenses.s3.amazonaws.com/tensorflow-${TF_VERSION}/license.txt -o /license.txt
```
Comment (tejaschumbalkar) [inline tensorflow/training/docker/2.9/py3/cu112/Dockerfile.gpu:250]: do we have the license file created for `tensorflow-2.9`?

File: tensorflow/training/docker/2.9/py3/Dockerfile.cpu [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+
+# Some TF tools expect a "python" binary
+RUN ln -s $(which ${PYTHON}) /usr/local/bin/python \
+ && ln -s $(which ${PIP}) /usr/bin/pip
+
+RUN apt-get update && apt-get -y install protobuf-compiler
+
+# install PyYAML==5.1.2 to avoid conflict with latest awscli
+# # python-dateutil==2.8.0 to satisfy botocore associated with latest awscli
+RUN ${PIP} install --no-cache-dir -U \
```
Comment (tejaschumbalkar) [inline tensorflow/training/docker/2.9/py3/Dockerfile.cpu:132]: Are we keeping the pinned versions of pip package as of TF2.8? Any modification required?

File: tensorflow/buildspec.yml [hunk 5]
```diff
@@ -39,65 +39,116 @@ context:
...
+    tag: !join [ *VERSION, "-", *DEVICE_TYPE, "-", *TAG_PYTHON_VERSION, "-", *CUDA_VERSION,
+      "-", *OS_VERSION, "-e3-example" ]
+    docker_file: !join [ docker/, *SHORT_VERSION, /, *DOCKER_PYTHON_VERSION, /example,
+      /Dockerfile., *DEVICE_TYPE ]
     enable_test_promotion: true
     context:
-      <<: *INFERENCE_CONTEXT
+      <<: *TRAINING_CONTEXT
+
+  # BuildTensorflowSageMakerGpuPy39Cu112TrainingDockerImage:
```
Comment (tejaschumbalkar) [inline tensorflow/buildspec.yml:93]: Missing E3 inference configurations in the commented section

File: tensorflow/training/docker/2.9/py3/cu112/Dockerfile.gpu (hunk 2 above)
Comment (nish21) [reply tensorflow/training/docker/2.9/py3/cu112/Dockerfile.gpu:199]: Updated

File: tensorflow/training/docker/2.9/py3/Dockerfile.cpu (hunk 4 above)
Comment (nish21) [reply tensorflow/training/docker/2.9/py3/Dockerfile.cpu:132]: Updated

File: tensorflow/buildspec.yml (hunk 5 above)
Comment (nish21) [reply tensorflow/buildspec.yml:93]: Since we don't have inference release from tensorflow yet, we could add it later with the PR for inference containers, instead of adding commented code. What do you think?

Comment (tejaschumbalkar) [reply tensorflow/buildspec.yml:93]: That works

File: test/dlc_tests/container_tests/bin/testTFKerasHVD.py (hunk 1 above)
Comment (nish21) [reply test/dlc_tests/container_tests/bin/testTFKerasHVD.py:33]: Confirmed that this change is compatible till at least TF2.5

**Review by tejaschumbalkar — APPROVED** [review]
